<launch>
    <arg name="world" default="worlds/empty.world"/>
    <arg name="start_pose" default="0 0 0"/>

    <param name="/use_sim_time" value="true" />

    <node name="gazebo" pkg="gazebo_ros" type="gzserver" respawn="false" output="screen"
          args="-e ode $(arg world)" required="false">
        <remap from="/camera/image_raw" to="/camera/image_rect"/>
    </node>

    <node name="gazebo_gui" pkg="gazebo_ros" type="gzclient" respawn="false" output="screen" required="true"/>

    <include file="$(find rr_platform)/launch/macaroni_state_publisher.launch"/>

    <node pkg="rr_platform" type="scanToPointCloud" name="scanToPointCloud" output="screen" />

    <node name="controller_spawner" pkg="rr_gazebo" type="controller_spawner.py" respawn="false" output="screen"
          args="joint_state_controller left_wheel_effort_controller right_wheel_effort_controller
            left_steer_position_controller right_steer_position_controller --shutdown-timeout 1"/>

    <rosparam file="$(find rr_gazebo)/config/macaroni_control.yaml" command="load"/>

    <node name="sim_controller" pkg="rr_gazebo" type="macaroni_controller" respawn="false" output="screen" />

    <node name="spawn_platform_script" pkg="rr_gazebo" type="spawn_macaroni_platform.sh" 
          args="$(arg start_pose)"/>
</launch>
